mesonbuild/compilers/compilers.py

```diff
@@ -1275,6 +1281,13 @@ def get_no_warn_args(self) -> T.List[str]:
     def needs_static_linker(self) -> bool:
         raise NotImplementedError(f'There is no static linker for {self.language}')
 
+    def analyzer_compile_args(self) -> T.List[str]:
+        if self.get_id() == 'gcc':
+            return ['-fanalyzer']
```

Comment:
The base compilers class should return `[]` and the method should be overridden in the gnu and clang mixins, rather than doing ID checks in the base class.

Reply:
Done the changes as part of latest commit. Thanks.